src/drt/src/dr/FlexDR_init.cpp

```diff
@@ -1579,68 +1608,43 @@ void FlexDRWorker::initTrackCoords_route(drNet* net,
       auto obj = static_cast<drPathSeg*>(uConnFig);
       const auto [bp, ep] = obj->getPoints();
       const auto lNum = obj->getLayerNum();
+      const auto lNum2 = initTrackCoords_getNonPref(lNum);
+      if (lNum2 < 0) {
+        std::cout << "Error: initTrackCoords cannot add non-pref track"
+                  << std::endl;
```

Comment:
I see you just moved this but following the boy scout rule it would be good to change this to use the logger and error() rather than std::cout.  You could move the check to initTrackCoords_getNonPref as there is nothing the caller is adding to the check.

Reply:
Using `error` would make it fatal and stop the process right ?
ATM (and in the previous code), although the message says "Error", it would still continue.
( The previous code was wrong because it would use 0 as lNum2 which is invalid for insertion in the map, here I'm not inserting anything at all for the non-pref direction if it wasn't found ).

But if stopping is the desired outcome, I can use `error` and then remove the extraneous checks.

Reply:
I say we make it a logger error and check if any of the current test cases break. The only case where this would trigger an error is if we are limited by routing on a single layer which is pretty unusual.

Reply:
Fixed, tests seem to pass.